script.js

```diff
@@ -1,4 +1,229 @@
+
+  if (
+    input != "Scissors" &&
+    input != "Paper" &&
+    input != "Stone" &&
+    input != "Reverse" &&
+    input != "Korean" &&
+    input != "Normal"
+  ) {
+    return "Your input is invalid";
+  }
+
+  if (gameMode == "Normal") {
+    if (
+      (computerChoice == "Scissors" && input == "Stone") ||
+      (computerChoice == "Paper" && input == "Scissors") ||
+      (computerChoice == "Stone" && input == "Paper")
+    ) {
+      userWinCount = userWinCount + 1;
+      var result = "You win!";
+    }
+
+    if (
+      (computerChoice == "Scissors" && input == "Paper") ||
+      (computerChoice == "Paper" && input == "Stone") ||
+      (computerChoice == "Stone" && input == "Scissors")
+    ) {
+      computerWinCount = computerWinCount + 1;
+      var result = "You lose!";
+    }
+
+    if (computerChoice == input) {
+      drawCount = drawCount + 1;
+      var result = "It is a draw!";
+    }
+  }
+
+  if (gameMode == "Reverse") {
+    if (
+      (computerChoice == "Scissors" && input == "Paper") ||
+      (computerChoice == "Paper" && input == "Stone") ||
+      (computerChoice == "Stone" && input == "Scissors")
+    ) {
+      userWinCount = userWinCount + 1;
+      var result = "You win!";
+    }
+
+    if (
+      (computerChoice == "Scissors" && input == "Stone") ||
+      (computerChoice == "Paper" && input == "Scissors") ||
+      (computerChoice == "Stone" && input == "Paper")
+    ) {
+      computerWinCount = computerWinCount + 1;
+      var result = "You lose!";
+    }
+
+    if (computerChoice == input) {
+      drawCount = drawCount + 1;
+      var result = "It is a draw!";
+    }
+  }
+
+  if (gameMode == "Korean") {
+    if (
+      (computerChoice == "Scissors" && input == "Stone") ||
+      (computerChoice == "Paper" && input == "Scissors") ||
+      (computerChoice == "Stone" && input == "Paper")
+    ) {
+      koreanWinner = "user";
+      return (
+        "You chose " +
+        input +
+        ". The computer chose " +
+        computerChoice +
+        ". Get a draw next to win!"
+      );
+    }
+
+    if (
+      (computerChoice == "Scissors" && input == "Paper") ||
+      (computerChoice == "Paper" && input == "Stone") ||
+      (computerChoice == "Stone" && input == "Scissors")
+    ) {
+      koreanWinner = "computer";
+      return (
+        "You chose " +
+        input +
+        ". The computer chose " +
+        computerChoice +
+        ". Get a draw next and the computer wins!"
+      );
+    }
+    if (computerChoice == input) {
+      if (koreanWinner == "user") {
+        koreanWinner = "";
+        userWinCount = +1;
+        return (
+          "Hi " +
+          userName +
+          "!<br><br>" +
+          computerChoiceMsg +
+          "<br><br>" +
+          yourChoiceMsg +
+          "<br><br>" +
+          "You win!<br>" +
+          userName +
+          " " +
+          userWinCount +
+          " - " +
+          computerWinCount +
+          " Computer"
+        );
+      }
```

Comment:
May want to have a separate function for handling the winning/losing statements as we see quite alot of repeated code here. 

It could be something like 

```
// make the function
var getWinningStatement = function (parameter1, parameter2, parameter3) {
    var ouputWinningStatment = 'Hi' + parameter1 + 'say some stuff' + parameter2 + 'more stuff' + parameter3;
    return ouputWinningStatment ;
} 

// call the function 
var winningStatement = getWinningStatement(a1,a2,a3); 

```